src/User/Password/PasswordAlgorithm.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MsgPhp\User\Password;
+
+/**
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+final class PasswordAlgorithm
+{
+    private const DEFAULT_LEGACY = 'sha512';
+
+    /** @var int|string */
+    public $type;
+
+    /** @var array|null */
+    public $options;
+
+    /** @var string|null */
+    public $salt;
+
+    /** @var bool */
+    public $legacy;
+
+    public static function create(int $type = \PASSWORD_DEFAULT, array $options = []): self
```

Comment:
The create methods should probably receive some documentation explaining the values. There's currently no way of telling what `$options` can contain or what the difference between int and string `$type` is.

Reply:
should be good now